### ISSUE TITLE
Better guidance on Stacking lock ups and timing 

### ISSUE THREAD

**Issue by GinaAbrams** [blockstack/stacks-wallet#551]
Can we clarify the following: if folks can or can't enroll stacking STX in the prepare phase of the *next* stacking cycle before one is up? Not sure if or where this is covered yet, but at a minimum sounds like something we should account for in the wallet UI.

**Comment by markmhx** [comment]
Is the idea that we add a note saying that users who lock STX for 2 or more cycles won't receive BTC for cycle 1 but will be qualified for BTC in subsequent cycles?

**Comment by GinaAbrams** [comment]
Correct me if I'm wrong here, but I thought I heard from stand up today that someone can't continuously stack from the same address. If they lock for x cycles with some STX. those STX would be locked until a time before the next x+1 cycle.

**Comment by markmhx** [comment]
Hmm I'm not sure what was mentioned exactly, but my understanding is that if a user locks N STX for X cycles, those N STX will remain locked until the very completion of the Xth cycle _plus_ any lead time between the moment of locking and start of the first cycle in which they participate.

For example:

- Each cycle is 1 week
- User locks 10 STX for 3 cycles at mid-point of currently running cycle
- 10 STX are locked for half a week without any chance to earn BTC since not yet participating
- At start of new cycle (after that half week), first of the 3 cycles begins
- User now has chance to earn BTC over course of 3 weeks
- At end of those 3 cycles, all 10 STX are unlocked immediately
- User's 10 STX have been locked for total of 3.5 weeks and can be re-locked for additional cycles (presumably missing out on the cycle that immediate follows the 3 cycles in which they did participate)

Does that make sense / help clarify?

**Comment by GinaAbrams** [comment]
I'm not concerned about the first .5 week, but that at the end of the 3 cycles, all 10 STX are unlocked immediately, but not in time to register for the 4th cycle immediately following.

**Comment by markmhx** [comment]
Got it, I think I understand the concern better now. So a note like:

> You will be able to initiate Stacking again once your STX unlocked after the chosen N cycles. However, you will not have the chance to participate in the immediately proceeding cycle given timing considerations. So please take that into consideration when determining the number of cycles to lock now.

@jasperjansz is working on improvements to the Stacking UX generally right now so can start incorporating something like this.

**Comment by markmhx** [comment]
That said, maybe they technically _can_ lock in the prepare cycle even if they're still finishing their last cycle for a previous lock. I'm assuming they can't with the above copy.

If they actually can, then we should simply free up the Stacking form again during that last cycle.

@agraebe any insight on how this works?

**Comment by GinaAbrams** [comment]
Awesome, yeah sorry it was hard for me to explain for a bit haha but think this would really help.

**Comment by markmhx** [comment]
All good, your explanation makes sense in hindsight. I confused myself as well =)

**Comment by agraebe** [comment]
💯 should explain this. there is no way to commit to an upcoming cycle while tokens are still locked. every stacker will need to go through a one-cycle waiting period in between stacking commitments.

**Comment by diwakergupta** [comment]
> there is no way to commit to an upcoming cycle while tokens are still locked. every stacker will need to go through a one-cycle waiting period in between stacking commitments.

The first sentence is accurate, but the 2nd sentence isn't entirely accurate. Keep in mind the prepare phase (100 blocks) and reward phase (2000 blocks) don't overlap. So if I've locked STX for N cycles, at the end of the Nth cycle, I have a window of 100 blocks to re-stack my STX.

**Comment by agraebe** [comment]
I double-checked this but from what I currently understand, the stacking commitment needs to happen _before_ the prepare phase.

**Comment by agraebe** [comment]
maybe @lgalabru or @kantai can chime in to ensure we're on the right track here

**Comment by kantai** [comment]
> I'm not concerned about the first .5 week, but that at the end of the 3 cycles, all 10 STX are unlocked immediately, but not in time to register for the 4th cycle immediately following. They would potentially miss a cycle of stacking by having to re-enroll in the prepare phase, unless they were stacking from multiple addresses at different times.

This is correct -- as PoX is currently implemented, a participant _must_ register for cycle "N" before cycle N starts. Any participant whose STX participated in cycle "N-1" would not be able to use those STX until cycle "N" began.

**Comment by diwakergupta** [comment]
> > I'm not concerned about the first .5 week, but that at the end of the 3 cycles, all 10 STX are unlocked immediately, but not in time to register for the 4th cycle immediately following. They would potentially miss a cycle of stacking by having to re-enroll in the prepare phase, unless they were stacking from multiple addresses at different times.
> 
> This is correct -- as PoX is currently implemented, a participant _must_ register for cycle "N" before cycle N starts. Any participant whose STX participated in cycle "N-1" would not be able to use those STX until cycle "N" began.

@kantai is this reflected in SIP-007 somewhere? Also, that last statement seems weaker than what others have expressed in this thread. Are these equivalent statements:

- STX participating in cycle "N-1" _can not_ re-stack until cycle "N+1"
- STX participating in prepare phase for cycle N can only stack starting cycle "N+1"

**Comment by kantai** [comment]
> @kantai is this reflected in SIP-007 somewhere?

Not very clearly -- SIP-007 does talk about how anything that depends on Stacks chain state _must_ query the PoX anchor block, and that the anchor block is always before the prepare phase begins (ideally, it's exactly the block before the prepare phase began, but that's what the prepare phase is reaching agreement on). 

> Also, that last statement seems weaker than what others have expressed in this thread. Are these equivalent statements:
>
>    STX participating in cycle "N-1" can not re-stack until cycle "N+1"
>    STX participating in prepare phase for cycle N can only stack starting cycle "N+1"

I don't know if they're equivalent statements, but they are both true in the current PoX implementation. An improvement to the `pox.clar` contract (and the locking behavior of the `stacks-node` in general) could allow an address currently locked to "extend" an active lock (improving on the situation in the first statement). However, with the way PoX anchor blocks works, no improvement could be made on the second statement.

**Comment by jasperjansz** [comment]
@kantai Question about this statement and the implications for the UI:

> STX participating in cycle "N-1" can not re-stack until cycle "N+1"

Does it mean the user can't lock their STX until cycle N+1, or does it mean they won't earn rewards until cycle N+1? If the former, we'd disable access to the stacking form. For the latter, we could enable it but add a disclaimer they won't be earning rewards until cycle N+1.

**Comment by agraebe** [comment]
It is the former. Payouts happen as blocks are mined. After a stacker is done with stacking, they have to wait one cycle. However, user can sign up for the next cycle while they wait. Not sure if the UI needs to be disabled.

I think the best is to call out that there will be a waiting period after the number of cycles one chooses to participate in. Selecting more cycles will help avoid missing some in the long run. That could be on the form where users add stacking details.

**Comment by aulneau** [comment]
I was rereading sip 7 over the weekend and this paragraph seems to conflict with what has been stated in this thread

> These messages may be broadcast either on the Stacks chain or the Bitcoin chain. If broadcast on the Stacks chain, these messages must be confirmed on the Stacks chain before the anchor block for the reward period. If broadcast on the Bitcoin chain, they may be broadcast during the prepare phase, but must be included before the prepare phase finishes.

**Comment by kantai** [comment]
> I was rereading sip 7 over the weekend and this paragraph seems to conflict with what has been stated in this thread

Yes, that line of SIP 7 is no longer correct -- all state changes must happen before the PoX anchor block is chosen.

**Comment by jasperjansz** [comment]
Propose to add a message to the last step of the stacking flow: "After your chosen duration you’ll have to wait one cycle before stacking from this address again"

https://www.figma.com/file/VcXbxT1EwXvE1nUlOiBEES/Wallet?node-id=4137%3A2
